src/dbus_api/tree.rs

```diff
@@ -857,6 +860,47 @@ impl DbusTreeHandler {
         }
     }
 
+    /// Send a signal indicating that the blockdev total physical size has
+    /// changed.
+    fn handle_blockdev_total_physical_size_change(
+        &self,
+        path: Path<'static>,
+        new_total_physical_size: Sectors,
+    ) {
+        let total_physical_size_prop =
+            blockdev_total_physical_size_to_prop(new_total_physical_size);
+        if let Err(e) = self.property_changed_invalidated_signal(
+            &path,
+            prop_hashmap!(
+                consts::BLOCKDEV_INTERFACE_NAME_3_3 => {
+                    Vec::new(),
+                    consts::BLOCKDEV_TOTAL_SIZE_PROP.to_string() =>
+                    box_variant!(total_physical_size_prop.clone())
+                },
+                consts::BLOCKDEV_INTERFACE_NAME_3_4 => {
+                    Vec::new(),
+                    consts::BLOCKDEV_TOTAL_SIZE_PROP.to_string() =>
+                    box_variant!(total_physical_size_prop.clone())
+                },
+                consts::BLOCKDEV_INTERFACE_NAME_3_5 => {
+                    Vec::new(),
+                    consts::BLOCKDEV_TOTAL_SIZE_PROP.to_string() =>
+                    box_variant!(total_physical_size_prop.clone())
+                },
+                consts::BLOCKDEV_INTERFACE_NAME_3_6 => {
+                    Vec::new(),
+                    consts::BLOCKDEV_TOTAL_SIZE_PROP.to_string() =>
+                    box_variant!(total_physical_size_prop)
+                }
+            ),
```

Comment:
I'm not sure this is the best thing to do here. Even if grow physical was added in r3, the total physical size property is available all the way back so theoretically even if a user is using a r3 method, we should notify them that the property in r1 (for example) has changed.

Reply:
I agree.